frontend/packages/console-shared/src/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { DraggableData, DraggableCore } from 'react-draggable';
+import { useSize } from './drawer-utils';
+import './Drawer.scss';
+
+type DrawerProps = {
+  height?: number;
+  defaultHeight?: number;
+  open?: boolean;
+  defaultOpen?: boolean;
+  maxHeight?: number | string;
+  isResizable?: boolean;
+  header: React.ReactNode;
+  headerClassName?: string;
+  onChange?: (open: boolean, height: number) => void;
+};
+
+const Drawer: React.FC<DrawerProps> = ({
+  children,
+  defaultHeight,
+  height,
+  maxHeight = '100%',
+  defaultOpen,
+  open = defaultOpen,
+  isResizable = false,
+  header,
+  headerClassName,
+  onChange,
+}) => {
+  const [heightState, setHeightState] = React.useState(defaultHeight);
+  const [lastObservedHeight, setLastObservedHeight] = React.useState<number>(null);
+  const [size, ref] = useSize<HTMLDivElement>();
+  const resizeHeight = height || heightState;
+  const minimumHeight = size.height || 0;
+  const handleDrag = (e: MouseEvent, data: DraggableData) => {
+    const newHeight = resizeHeight - data.y;
+    if (newHeight < 0) return;
+    if (onChange) {
+      onChange(open, newHeight);
```

Comment:
You need to notify open and closed state as well. If closed, send last observed height too.

Reply:
If we send the lastObservedheight while closed then there will be fluctuation once it is moved in the open state because it has the height from the parent component from last onChange call